### ISSUE TITLE
HTML in headers rendered inconsistently

### ISSUE THREAD

**Issue by winniehell** [danielgtaylor/aglio#159]
Using aglio `2.0.4`, the following
```markdown
## App <span>foo</span>
```
leads to
![toc](https://cloud.githubusercontent.com/assets/1652293/9546306/f1be92d8-4d91-11e5-90b6-d48b3d0f2995.PNG)
in the table of contents but to
![doc](https://cloud.githubusercontent.com/assets/1652293/9546313/fd2d6a72-4d91-11e5-809b-fe29af159cb8.PNG)
in the document itself.

I'd prefer having the HTML not escaped in the table of contents allowing e.g. custom badges. In my opinion, it should at least be consistent between table of contents and document whether HTML is escaped or not.

**Comment by danielgtaylor** [comment]
Thanks for catching this!

**Comment by winniehell** [comment]
Additionally, the generated anchors may cause problems. I am not sure what a desired behavior would be:

```markdown
## App <span>foo</span>
```
may lead to `#header-app-foo` or `#header-app`.

For example,
```markdown
## <strong>Important</strong> Settings
```
should clearly be `#header-important-settings` but noone wants
```markdown
## <i class="material-icons">settings_input_component</i> Settings
```
to become `#header-settings_input_component-settings`.

The second case however may be solved by additional syntax - there is a [CommonMark discussion] and [kramdown] already supports it.

[CommonMark discussion]: http://talk.commonmark.org/t/consistent-attribute-syntax/272/95
[kramdown]: http://kramdown.gettalong.org/quickref.html#block-attributes

**Comment by danielgtaylor** [comment]
I agree that the generated header names with HTML are a difficult problem and I've approached them as best I could. Definitely open to suggestions, but please open a new issue.

The originally reported issue is now fixed in the 1.2.1 release of the Olio theme engine.

**Comment by winniehell** [comment]
This is awesome! :heart: 

The following
```markdown
# Group <strong>Important</strong> Stuff

## Bar <span class="badge"><i class="fa fa-plane"></i></span>

```
now renders like this in the table of contents:
![toc](https://cloud.githubusercontent.com/assets/1652293/9614173/5f87c36e-50f0-11e5-8161-b0257c171e73.png)

So you can now use custom badges very easily. :+1: 

Sorry, the second comment was just random thoughts while I was not in office. I'll open another issue.
